mlflow/deployments/cli.py

```diff
@@ -258,3 +258,24 @@ def predict(target, name, input_path, output_path):
             predictions_to_json(result, fp)
     else:
         predictions_to_json(result, sys.stdout)
+
+
+@commands.command("explain")
+@deployment_name
+@target_details
+@parse_input
+@parse_output
+def explain(target, name, input_path, output_path):
+    """
+    predict the features importance for the deployed model for the given input(s)
+    """
+    import pandas as pd
+
+    df = pd.read_json(input_path)
+    client = interface.get_deploy_client(target)
```

Comment:
Just a thought:  Should we raise an exception here if target is not `torchserve` ?

Reply:
@shrinath-suresh we probably don't want to condition logic in MLflow core on specific plugins that exist outside MLflow (you could imagine then we'd have a big branching if-statement for all deploy plugins/in order to add a new plugin folks would have to know to merge a PR modifying this file with details of their own plugin, vs being able to develop & release a plugin independently of MLflow.

Could we instead check whether the `explain` method is defined on the `client` object and raise an exception saying that the current plugin doesn't support the `explain` API if not? e.g. something like

```suggestion
    client = interface.get_deploy_client(target)
     if not hasattr(client, "explain"):
         raise MlflowException("Computing model explanations is not yet supported for deployment target %s" % target)
     result = client.explain(name, df)
     ... 
```

Reply:
If we add explain method as non-abstract method in base deployment class then "hasattr(client, "explain")" will always return True. So, Can we add the exception in the base class's explain method itself?

```
    def explain(self, deployment_name, df):
        raise MlflowException("Computing model explanations is not yet supported for this deployment target")
```

Reply:
@KasirajanA yep that makes sense!

Reply:
And I think adding it as a non-abstract method (good catch there as well) should allow us to make this change without breaking existing plugins, as you described :)